sssd_test_framework/roles/ad.py

```diff
@@ -1490,4 +1500,309 @@ def __get_info(
+        """
+        Set group policy SeRemoteInteractiveLogonRight or SeDenyRemoteInteractiveLogonRight key.
+
+        Create two lists, allow and deny. Iterate through the user dictionary, get the value for SSH
+        The dictionary keys will be the following, name: str, ssh: bool, su: bool
+        """
+        allow: list = ["Administrators"]
+        deny: list = []
+
+        # iterate through dictionary, if ssh is True, permit, if ssh is False, deny.
+
+        self._set(f"SeRemoteInteractiveLogonRight", ";".join(allow))
+        self._set(f"SeDenyRemoteInteractiveLogonRight", ";".join(deny))
+
+    def set_su(self, users: list[dict[str, bool, bool]]) -> None:
+        """
+        Set group policy SeInteractiveLogonRight or SeDenyInteractiveLogonRight key.
+        """
+        allow: list = ["Administrators"]
+        deny: list = []
+
+        # iterate through dictionary, if su is True, permit, if su is False, deny.
+
+        self._set(f"SeInteractiveLogonRight", ";".join(allow))
+        self._set(f"SeDenyInteractiveLogonRight", ";".join(deny))
+
+
+class GPPermissions(GPObject):
+    """
+    Group Policy permissions management.
+    """
+
+    def __init__(self, role: AD, name: str) -> None:
+        """
+        :param role: AD role object.
+        :type role: AD
+        :param name: Group Policy name.
+        :type name: str
+        """
+        super().__init__(role, name)
+
+    def set(
+        self,
+        *,
+        target_name: str,
+        target_type: str | None = "group",
+        permissions: str | None = "GpoRead",
+    ) -> None:
+        """
+        Set the Group Policy permissions.
+
+        :description:"GpoRead", "GpoApply", "GpoEdit", "GpoEditDeleteModifySecurity", "None"
+
+        :type target_name: str
+        :param target_type: Type of target, "group" or "user", defaults to "group"
+        :type target_type: str
+        :param permissions: Permission level, defaults to GpoRead
+        :type permissions: str | optional
+        """
+        args: CLIBuilderArgs = {
+            "TargetName": (self.cli.option.VALUE, target_name),
+            "TargetType": (self.cli.option.VALUE, target_type),
+            "PermissionLevel": (self.cli.option.VALUE, permissions),
+        }
+
+        self._add("Set-GPPermission", **self._name, **args)
+
+
+class GPO(GPObject):
+    """
+    Group Policy Object management.
+    """
+
+    def __init__(
+        self,
+        role: AD,
+        name: str,
+        target: str | None = "Default-First-Site-Name",
+    ) -> None:
+        """
+        :param role: AD host object.
+        :type role:  ADHost
+        :param name: Group policy name.
+        :type name: str
+        :param target: Group policy target.
+        :type target: str
+        """
+        super().__init__(role, name, target)
+
+        self.permissions: GPPermissions = GPPermissions(role, self.name)
+
+        self.key: GPKey = GPKey(role, self.name)
+
+    def link(self) -> None:
```

Comment:
I would move target as a parameter of link instead of setting it in constructor.